src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -357,12 +357,10 @@ class ClusterMonitorActor(val cluster: Cluster,
   private def getClusterInstances: Future[Set[Instance]] = {
     for {
       map <- gdDAO.getClusterInstances(cluster.googleProject, cluster.clusterName)
-      instances <- Future.traverse(map) { case (role, instances) =>
-        Future.traverse(instances) { instance =>
-          googleComputeDAO.getInstance(instance).map(_.map(_.copy(dataprocRole = Some(role))))
-        }
+      instances <- map.toList.flatTraverse { case (role, instances) =>
```

Comment:
`Future.traverse` doesn't work for `Map` with `partial-unification` on. Hence refactor a bit